src/components/document/SubTabs/SubTabs.tsx

```diff
@@ -14,16 +15,31 @@ interface Tab {
     tag: string
     name: string
     icon: ReactElement
-    content: ReactElement
+    content: string
 }
 
 export const SubTabs = observer(() => {
     const { classes } = useStyles()
     const {
         printMode,
-        documentStore: { data, ocrData, collection, subTab, handleSubTabChange },
+        documentStore: {
+            data,
+            ocrData,
+            collection,
+            subTab,
+            handleSubTabChange,
+            documentSearchStore: { query, setActiveSearch, textSearchStore },
+        },
     } = useSharedStore()
 
+    useEffect(() => {
+        setActiveSearch(textSearchStore)
+    }, [setActiveSearch, textSearchStore])
+
+    useEffect(() => {
+        textSearchStore.setSubTab(subTab)
+    }, [subTab, textSearchStore])
```

Comment:
use reaction on `documentStore.subTab` change instead of these effects

Reply:
it should be pretty easy to change indexes to string names, the main idea it to get rid of `useEffect` hooks because it makes code complex, hard to read, understand and maintain

https://mui.com/material-ui/api/tab/

value | any |   | You can provide your own value. Otherwise, we fallback to the child position index.
-- | -- | -- | --

Reply:
I did not want to refactor the code to much for just 3 useEffects. We need to change the entire structure of how tabs are stored and used from the URL, in order to navigate through all of the tabs that we have, pdf, text, meta, tags, location, and so on. I'm not saying its complicated, but it would take additional, time. Time that we don't have until tomorrow, for the deployment and testing. 

Since this is working as it is, with these 3 useEffects which set the active search. We can leave it like this for now, and create a separate ticket to refactor the Tab and subtab navigation. 

I do agree with you, that this is better if we do it like this, refactor the tabs and subtabs and also create reactions for the search stores as well.

Reply:
ok, let's keep it as a temporary solution